### PR TITLE
Stacksplit

### PR DESCRIPTION
**Pull request by hamzah-hayat**
New Alt Click ability on stacks
Lets you split stacks into different size piles using a number

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -225,19 +225,44 @@
...
+		//get amount from user
+		var/min = 0
+		var/max = src.get_amount()
+		var/stackmaterial = input(user,"How many sheets do you wish to take out of this stack? (Maximum  [max]") as num
+		if(stackmaterial == null || stackmaterial <= min || stackmaterial >= src.get_amount())
+			return
+		else
+			change_stack(user,stackmaterial)
+			user << "<span class='notice'>You take [stackmaterial] sheets out of the stack</span>"
+	return
```
Comment (RemieRichards) [inline code/game/objects/items/stacks/stack.dm]: trailing return, remove it please.

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -225,19 +225,44 @@
...
+
+/obj/item/stack/proc/change_stack(mob/user,amount)
+	var/obj/item/stack/F = new src.type(user, amount)
+	. = F
+	F.copy_evidences(src)
+	user.put_in_hands(F)
+	src.add_fingerprint(user)
```
Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: no need for src here

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -225,19 +225,44 @@
...
+	F.add_fingerprint(user)
+	use(amount)
+	if (src && usr.machine==src)
```
Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: at what point could this possibly disappear during this proc?

**Comment by chickenboy10** [comment]
I've made the changes based on your feedback, but im a bit wary of changing the code inside the change stack proc because it was written before I got to it and I dont want to break anything by making unnecessary changes.

**Pull request merged**